src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp

```diff
@@ -740,6 +741,70 @@ std::set<std::vector<element::Type>> jit_gelu_tanh_emitter::get_supported_precis
     return {{element::f32}};
 }
 
+/// GELU_V7 ///
+jit_gelu_v7_emitter::jit_gelu_v7_emitter(dnnl::impl::cpu::aarch64::jit_generator* host,
+                                         dnnl::impl::cpu::aarch64::cpu_isa_t host_isa,
+                                         const std::shared_ptr<ov::Node>& node)
+        : jit_emitter(host, host_isa, node, get_arithmetic_binary_exec_precision(node)) {
+    const auto gelu = std::dynamic_pointer_cast<ov::op::v7::Gelu>(node);
+    if (gelu == nullptr) {
+        OV_CPU_JIT_EMITTER_THROW("Can't cast to ov::op::v7::Gelu");
+    }
+    approximationMode = gelu->get_approximation_mode();
+    if (approximationMode == ov::op::GeluApproximationMode::ERF) {
+        gelu_erf_emitter = std::make_unique<jit_gelu_erf_emitter>(h, host_isa, node);
+    } else if (approximationMode == ov::op::GeluApproximationMode::TANH) {
+        gelu_tanh_emitter = std::make_unique<jit_gelu_tanh_emitter>(h, host_isa, node);
```

Comment:
The present design of gelu emitters requires a bunch `if`s, e.g. here, on L766, L772, L787, L793, L799 etc
This increases possibility of errors, complicates debugging, reduces readability and so on.
So here are some suggestions on how we can improve it:
1) We can move this selection logic inside the [v7::Gelu emitter builder](https://github.com/openvinotoolkit/openvino/blob/443cd98f10e362baf941003ef26caeb3888c32c6/src/plugins/intel_cpu/src/emitters/snippets/aarch64/cpu_generator.cpp#L104), so the builder would instantiate the correct emitter (erf or tanh) right away.
2) If we the emitters can share some common code & functionality (not sure about it, but still), we can rename `jit_gelu_v7_emitter` to `jit_gelu_base_emitter` and derive `jit_gelu_erf_emitter` and `jit_gelu_tanh_emitter` from it to maximize code reusage.
This way we won't need all these `if`'s. What do you think?

Reply:
Good point! Thanks Ivan! Based on your suggestion, I added a base class jit_gelu_base_emitter. Now those `if` conditions are gone. 

As far as I understand, the builder needs key-value pairs. For both approximationMode in v7::Gelu, the keys are the same ov::op::v7::Gelu::get_type_info_static(). So it seems we can not distinguish the two modes when registering them. So I simply added a base class `jit_gelu_base_emitter`, and Let jit_gelu_v7_emitter holder a pointer to the base class. Then we can get rid of all the `if` conditions by utilizing polymorphism. Please feel free to have further discussions or give more specific suggestions, if you have concerns. Thanks a lot!

Reply:
> As far as I understand, the builder needs key-value pairs

It's not entirely so. Builder itself is simply a [lambda](https://github.com/openvinotoolkit/openvino/blob/c880ea9584b4b7910e217d2c180f4adb6ffc9f15/src/plugins/intel_cpu/src/emitters/snippets/aarch64/cpu_generator.cpp#L33-L35) that takes `ExpressionPtr` and returns `std::shared_ptr<snippets::Emitter>`. 

You are right that this builder lambda (paired with [another lambda](https://github.com/openvinotoolkit/openvino/blob/80eb61a85491081bd49bcc030a3602635b600924/src/common/snippets/include/snippets/target_machine.hpp#L26-L27) for precision reporting) is mapped to type_info_static. But it doesn't really matter, since we can hide this [emitter selection logic](https://github.com/openvinotoolkit/openvino/blob/b83a63dfe6ba92bc4240324a54ff694d6e75244d/src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp#L766-L770) inside the builder-lambda anyway.
So yes, both gelu modes will be mapped to the same builder lambda, but inside the lambda we can cast the passed node to gelu, get the approximation mode and based on that create the appropriate emitter (so basically everything you do in the jit_gelu_v7_emitter constructor).
The main advantage of this approach is that we won't need a base class for two jit gelu emitters and won't need to static_cast the pointers as you do now.
The only drawback is that we'll have to create a custom lambda-builder, instead of just using the one under marco define, but I think we can live with that 🙂

Reply:
Applied. Thanks Ivan for the detailed suggestions!